pkg/pool-stable/contracts/StablePool.sol

```diff
@@ -171,7 +171,7 @@ contract StablePool is IStablePool, BalancerPoolToken, BasePoolAuthentication, P
     }
 
     /// @inheritdoc IBasePool
-    function onSwap(PoolSwapParams memory request) public view onlyVault returns (uint256) {
+    function onSwap(PoolSwapParams memory request) public view virtual onlyVault returns (uint256) {
```

Comment:
Should we make this (and WeightedPool) non-view here to enable local state-changing operations in derived pools?

Reply:
not really; that will trigger a warning and it's just not worth it.